x-pack/legacy/plugins/apm/public/components/app/Settings/CustomizeUI/CustomActionsOverview/CustomActionsFlyout/FiltersSection.tsx

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSelect,
+  EuiSpacer,
+  EuiText,
+  EuiTitle
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { isEmpty } from 'lodash';
+import React, { useRef } from 'react';
+
+export interface Filter {
+  key: string;
+  value: string;
+}
+
+const DEFAULT_OPTION = {
+  value: 'DEFAULT',
+  text: i18n.translate(
+    'xpack.apm.settings.customizeUI.customActions.flyOut.filters.defaultOption',
+    { defaultMessage: 'Select fields...' }
+  )
+};
+
+const filterOptions = [
+  DEFAULT_OPTION,
+  { value: 'service.name', text: 'service.name' },
+  { value: 'service.environment', text: 'service.environment' },
+  { value: 'transaction.type', text: 'transaction.type' },
+  { value: 'transaction.name', text: 'transaction.name' }
+];
+
+export const FiltersSection = ({
+  filters = [{ key: '', value: '' }],
+  onChange
+}: {
+  filters: Filter[];
+  onChange?: (filters: Filter[]) => void;
```

Comment:
Is there any reason for not making it either required or removing it?
afaict it's not used anywhere.

Reply:
It's used to update the filters in the State. I made it optional because it's handled automatically by `react-hook-form`.
```js
<Controller
  as={<FiltersSection filters={filters} />}
  name="filters"
  control={control}
/>
```
But I can change it to required and add a @ts-ignore to avoid the warning TS throws.

Reply:
Ah okay. No, just leave it as-is but add a comment where `onChange` is coming from

Reply:
When I think about it, I'm starting to wonder whether using this kind of abstraction is worth it. 
Is is possible to use less of react-hook-form and still get value out of it? I'd like to avoid that having `react-hook-form` specific knowledge is required to work on the APM app.

WDYT? Would this be significantly harder or more verbose to do without `react-hook-form`? Or without the `Controller` part?

Reply:
It won't be that much difference @sqren, `react-hook-form` would make a difference if we had a lot of validations in the form. But in our case, we only check required fields. I will create a version without it, so we can compare and make a decision.

Reply:
> I will create a version without it, so we can compare and make a decision.

Thanks! I think this extra work will pay off down the line.